external/java/src/main/java/ch/epfl/dedis/byzcoin/ByzCoinRPC.java

```diff
@@ -686,6 +686,24 @@ public static Darc makeGenesisDarc(Signer admin, Roster roster) throws Cothority
         return d;
     }
 
+    /**
+     * Asks the roster for all the existing ByzCoin IDs.
+     *
+     * @param roster list of all cothority servers with public keys
+     * @return a list of SkipblockIds which are have a ByzCoin service running
+     * @throws CothorityCommunicationException if the communication fails or the message is corrupted
+     */
+    public static List<SkipblockId> getAllByzCoinIDs(Roster roster) throws CothorityCommunicationException {
+        ByzCoinProto.GetAllByzCoinIDsRequest.Builder b = ByzCoinProto.GetAllByzCoinIDsRequest.newBuilder();
```

Comment:
I'm not sure how much this must be backward compatible but calling conode running 3.1.1 will fail as the message doesn't exist.

Reply:
Well, the go client API also has this problem if you put it that ways: https://github.com/dedis/cothority/blob/master/byzcoin/api.go#L94

I think it's ok because old clients will continue to work when using the old APIs on new conodes.

Reply:
Oh right the fix was at the CLI level. Ok that's fair.